src/txn/txn_rollback_to_stable.c

```diff
@@ -251,12 +251,19 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
         WT_ERR(hs_cursor->get_value(
           hs_cursor, &hs_stop_durable_ts, &hs_durable_ts, &type_full, hs_value));
         type = (uint8_t)type_full;
-        if (type == WT_UPDATE_MODIFY)
-            WT_ERR(__wt_modify_apply_item(
-              session, S2BT(session)->value_format, &full_value, hs_value->data));
-        else {
-            WT_ASSERT(session, type == WT_UPDATE_STANDARD);
-            WT_ERR(__wt_buf_set(session, &full_value, hs_value->data, hs_value->size));
+
+        /*
+         * Do not include history store updates greater than on-disk data store version to construct
+         * a full update to restore.
+         */
+        if (hs_start_ts < unpack->tw.start_ts) {
```

Comment:
Does this work with non-timestamped updates?

Reply:
Yes. Non-timestamped tables we don't perform the rollback to stable, because the latest data is stable according to the transaction id. But this check needs to be enhanced when the RTS support for checkpoint snapshot is added.

Reply:
In the PR pf WT-6504, I found an issue that hs_start_ts which is from the hs key is not zero but unpack->tw.start_ts is zero because when the ondisk page is reconciled, the timestamp is older than the global oldest timestamp. So this comparison isn't correct sometimes. We need to do something else.

Reply:
OK. But the issue which you mentioned cannot occur with RTS. If the on-disk update has zero timestamps, we never rollback it. Any way to protect such cases when RTS with checkpoint snapshot is added, I will change it to compare with the `cbt->upd_value->tw.start_ts` instead of `hs_start_ts`

Reply:
I don't think changing to tw in cbt solves the issue. Basically it falls back to when the data page and hs page are reconciled. They may write inconsistent timestamps to disk as well.

Reply:
The data store gets reconciled first and then the history store reconciled for a page data that is moved. In your case, the comparison is done against history store key values. If you use the cbt->upd_value time window, there shouldn't be a problem.